pbj-integration-tests/src/main/java/com/hedera/pbj/integration/fuzz/ObjectBuilder.java

```diff
@@ -0,0 +1,188 @@
+package com.hedera.pbj.integration.fuzz;
+
+import com.hedera.pbj.runtime.EnumWithProtoMetadata;
+import com.hedera.pbj.runtime.OneOf;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+/**
+ * A factory for Protobuf model objects.
+ *
+ * The implementation uses Java Reflection to fetch a constructor,
+ * assuming there's only a single one, and then builds a list of
+ * argument values using random numbers generator. The arguments
+ * list branches for every found OneOf field.
+ *
+ * The factory returns a list of random objects for every branch.
+ *
+ * For example, an
+ *
+ *     AccountID(long shardNum, long realmNum,
+ *         OneOf<AccountID.AccountOneOfType> account)
+ *
+ * will yield 2 random objects, one for each branch declared in
+ * the AccountOneOfType enum.
+ *
+ * A support for field types is currently limited to just a few types.
+ * We will be adding support for more field types as we use this
+ * factory for more and more models. The factory throws an exception
+ * when it encounters an unsupported type, so it's easy to detect this
+ * and then add support for new types in the future.
+ */
+public class ObjectBuilder {
+    private static final Random RND = new Random();
+
+    /**
+     * Takes a Protobuf model class as an argument and return a list
+     * of random objects of this class for every possible OneOf branch.
+     * If the class doesn't have a single OneOf field, then a list
+     * with a single random object is returned.
+     *
+     * A random object is defined as an object obtained by calling
+     * the class' constructor and passing random values for its
+     * arguments.
+     */
+    public static <T> List<T> build(Class<T> clz) {
+        try {
+            Constructor<?>[] constructors = clz.getConstructors();
```

Comment:
Rather than to use Java reflection APIs, did you consider using builders, which are generated for every proto type? For example, `AccountID.newBuilder()`.

Reply:
My original sample test (that was just a toy test) used a builder for the AccountID. After seeing the number of models that we'd want to test and the number of variants of those models (based on their OneOfs), I realized that this would require too much of manual work to write the builder calls and then support them as we add/modify models.

So I opted for a mechanism that does this automatically. With this approach the only thing that would be maintained manually is a list of classes that we want to test (and even that can be automated by finding relevant classes in the classpath if we want in the future.)

An approach with generating code for calling builders would require an extra step of having to compile it before it can be used, and I find this approach more complex than the ObjectBuilder approach in this fix.

Reply:
Please see the other thread. I now use the ARGUMENTS from test models.